### PR TITLE
Add Java code examples [ECR-4250]

### PR DESCRIPTION
**Pull request by dmitry-timofeev**
### Overview

Adds a plugin to include Java code examples from source files.

Adds Exonum Java Binding as a Git submodule, to be able to include the code snippets from its source.

### Questions
1. Shall we use a git submodule or add a script syncing particular directories from EJB repository? Potentially, it will allow to reduce the paths lenghts, but will add 'accidental' changes to the diff (changes that have already been reviewed during the modifications in the main repository).

### See Also
- ECR-4250
- ECR-4132
- ECR-4258

### CODE REVIEW

**Comment by dmitry-timofeev** [comment]
> DEPRECATION: Python 2.7 will reach the end of its life on January 1st, 2020. Please upgrade your Python as Python 2.7 won't be maintained after that date. A future version of pip will drop support for Python 2.7. More details about Python 2 support in pip, can be found at https://pip.pypa.io/en/latest/development/release-process/#python-2-support

🙀 

Is there a reason we use Python 2?

**Comment by dmitry-timofeev** [comment]
I successfully served the docs locally with 
```
(.venv) dmitry-mac:exonum-doc user$ python --version
Python 3.7.6
```

**Comment by dmitry-timofeev** [comment]
Here is a result:

изображение

File: src/get-started/HelloWorld.java [hunk 1]
```diff
@@ -0,0 +1,7 @@
+
+public final class Run {
```
Comment (bullet-tooth) [inline src/get-started/HelloWorld.java]: why do we need this?

Comment (dmitry-timofeev) [reply src/get-started/HelloWorld.java]: Well, it is a prototype: https://github.com/exonum/exonum-doc/pull/403/commits/b02a6ba964d18d046a94a9e8efd56313fb9f044a

File: README.md [hunk 2]
```diff
@@ -99,6 +99,14 @@ npm install
 
 (you will need Node 8+ installed).
 
+<!-- Todo: More appropriate place? -->
```
Comment (vitvakatu) [inline README.md]: Actually, I think we need to add an instruction with
```
git clone https://github.com/exonum/exonum-doc
cd exonum-doc
git submodule update --init --recursive
```
somewhere at the top.

**Comment by dmitry-timofeev** [comment]
I updated the code, it it no longer a prototype: ECR-4258

File: README.md [hunk 3]
```diff
@@ -21,6 +21,13 @@ You can read about Markdown [here](https://guides.github.com/features/mastering-
 or [other](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)
 [places](http://www.mkdocs.org/user-guide/writing-your-docs/).
 
+Code examples are included from source using 
+[`mkdocs-codeinclude-plugin`][codeinclude-plugin] to ensure they are
+always up-to-date and correct.
+
+<!-- todo: Replace the link to our fork with the upstream link once our PRs are integrated. -->
```
Comment (bullet-tooth) [inline README.md]: add here PR links to do not search them in future

Comment (dmitry-timofeev) [reply README.md]: There are about 4 of them: https://github.com/rnorth/mkdocs-codeinclude-plugin/pulls

**Comment by dmitry-timofeev** [comment]
@slowli 

How do we exclude the sources from analyses? I think we can 
- add exclude options to each tool in use (if all of them have such options)
- move the code-examples out of `src`

**Comment by slowli** [comment]
@dmitry-timofeev 

> move the code-examples out of `src`

I'd suggest to try this version first.

**Comment by dmitry-timofeev** [comment]
It worked, Travis is happy.

**Comment by dmitry-timofeev** [comment]
So, do we need to install it in `edit` mode or not?

**Comment by dmitry-timofeev** [comment]
Discussed to install it in edit mode in `3rd-party` so that it does not get onto the site.

File: README.md [hunk 4]
```diff
@@ -93,8 +93,9 @@ git submodule update --init --recursive
 First, you need to install [Python](http://python.org/) and [python-pip](https://pip.readthedocs.io/en/stable/installing/).
 Then, install the `mkdocs` theme together with its dependencies:
 
+<!-- Remove 3rd-party in ECR-4265 -->
 ```
-pip install -r requirements.txt
+pip install -r requirements.txt --src 3rd-party
```
Comment (dmitry-timofeev) [inline README.md]: @atrykush 

Does the Jenkins job publishing the docs need to be updated?

**Pull request merged**